src/users/dummy.py

```diff
@@ -0,0 +1,45 @@
+"""Dummy adapter for communicating with Keycloak to be used for testing."""
+from asyncio import Lock
+from typing import Dict, Optional
+from uuid import uuid4
+
+from src import models
+
+
+class DummyUserStore:
+    """A dummy adapter for keycloak. By default it will create and return users that do not exist."""
+
+    def __init__(self, *, user_always_exists: bool = True):
+        self._users: Dict[str, models.User] = {}
+        self._lock = Lock()
+        self._user_always_exists = user_always_exists
+
+    @property
+    def user_always_exists(self) -> bool:
+        """If true then every query for a user will make a user that matches the query."""
+        return self._user_always_exists
+
+    @user_always_exists.setter
+    def user_always_exists(self, value: bool):
+        """Control over the users that are queried should exist or not."""
+        self._user_always_exists = value
```

Comment:
since the setter doesn't do anything special, can this just be a regular @property?

Reply:
@Panaetius by regular `@property` you mean not an `@property` but rather just simply have `self.user_always_exists`, right? Because if I remove the setter then I cannot change the user store to give me a 404 when testing via the setter. And I will want to do this when running tests. That is the main reason for this. This would not be used in production only for testing or development.

```python
class DummyUserStore:
    """A dummy adapter for keycloak. By default it will create and return users that do not exist."""

    def __init__(self, *, user_always_exists: bool = True):
        self._user_always_exists = user_always_exists

    @property
    def user_always_exists(self) -> bool:
        """If true then every query for a user will make a user that matches the query."""
        return self._user_always_exists
```

Reply:
Let me know what you think. Either way this is really inconsequential because it will never run in production.

Reply:
I meant 
```
class DummyUserStore:
    user_always_exists = True

    def __init__(self, *, user_always_exists: bool = True):
        self.user_always_exists = user_always_exists

```

Since there's not really anything gained by having setter/getters. But as you say, it doesn't really matter :slightly_smiling_face:

Reply:
Ok got it. I can change that.